### PR TITLE
[DataGrid] Remove `line-height` from `GridCell`

### LINKED ISSUE #2859: [DataGrid] Multiple lines/children in renderCell doesn't work for MUI5
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Multiple lines/children in cell doesn't work for MUI5 (example of solution on [stackoverflow](https://stackoverflow.com/questions/64865959/with-material-uis-datagrid-component-how-do-i-create-multi-layered-text-in-one) works fine for `MUI4`, but not for `MUI5`);

### Expected behavior 🤔

Multiple lines/children in cell works well

### Steps to reproduce 🕹

Steps:

1. Open [codesandbox](https://codesandbox.io/s/autumn-currying-kwxef?file=/src/Demo.tsx) example and see result
2. `Devtools` disabling `lineHeight` solve this issue, but I can't find any workarounds to do it through the code (probably it assigned/calculated by scripts, mui devs knows better)

image
image

### Context 🔦

This issue blocks me. I need to render array of elements in a cell (with my own component with scroller); example of structure:
```
type Metrics = {
  key: string; 
  value: number;
};

type ServerResponse = {
  // some fields from response
  metrics: Metrics[];
}
```
And I need to show that metrics array for each entity (row) in a single cell

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Browsers: Chrome
```
image
</details>

### Order ID 💳 (optional)

_No response_

### PR DESCRIPTION
**Pull request by DanailH**
Fixes #2859 

A small change to the way we center content in grid cells - the result is to enable multi-lines in grid cells.

### CODE REVIEW

**Comment by DanailH** [comment]
I marked Argos as approved - the difference was in the font.

To summarise - I removed the `line-height` because it was not possible to have multiline content in cells. You can increase the height of the cell but you can't make the text go on 2 lines for example.

The change is that now the cell is always` display: flex` and text positioning is done via `justify-content`.
Technically the `cell--withRenderer`class can be depricated. I left it a slot to avoid breaking change but to also give the ability for users to still be able to only target cells with custom components in them.

File: packages/grid/_modules_/grid/components/containers/GridRootStyles.ts [hunk 1]
```diff
@@ -254,34 +255,18 @@ export const GridRootStyles = styled('div', {
       alignItems: 'center',
       gridGap: theme.spacing(1),
     },
-    [`& .${gridClasses['cell--withRenderer']}`]: {
```
Comment (m4theushw) [inline packages/grid/_modules_/grid/components/containers/GridRootStyles.ts:257]: Could you update https://github.com/mui-org/material-ui-x/blob/8013d30a182139b6c1d87b23ff664f22cbf9fba3/packages/grid/_modules_/grid/gridClasses.ts#L318 and mark this class name with `// TODO v6 - remove`?

Comment (DanailH) [reply packages/grid/_modules_/grid/components/containers/GridRootStyles.ts:257]: added

**Review by m4theushw — APPROVED** [review]

File: packages/grid/_modules_/grid/components/containers/GridRootStyles.ts [hunk 2]
```diff
@@ -213,7 +214,8 @@ export const GridRootStyles = styled('div', {
       },
     },
     [`& .${gridClasses.cell}`]: {
-      display: 'block',
+      display: 'flex',
+      alignItems: 'center',
       overflow: 'hidden',
       textOverflow: 'ellipsis',
       whiteSpace: 'nowrap',
```
Comment (m4theushw) [inline packages/grid/_modules_/grid/components/containers/GridRootStyles.ts:221]: The `textOverflow` and `whiteSpace` do nothing on flexbox. Related to #3587

**Pull request merged**